### PR TITLE
Adding C++ in experimental and Kotlin as Beta

### PR DESCRIPTION
**Pull request by aryx**
test plan:
yarn start
see updated columns in top page and supported-language page

### Security

- [x] Change has no security implications (otherwise, ping the security team)

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -45,12 +45,14 @@ Semgrep is developed and commercially supported by [r2c, a software security com
 
 Semgrep supports 17+ languages.
 
+<!--  coupling: if you modify this table, modify also status.md -->
+
 <div id="language-support-table">
 
 | GA ✅      | Beta 🐛                     | Experimental 🚧            |
 |:---------- |:---------------------------|:---------------------------|
 | C#         | Terraform                  | C                          |
```
Comment (pabloest) [inline docs/index.md:54]:
```suggestion
| C#         | Kotlin                     | C                          |
```

File: docs/index.md [hunk 2]
```diff
@@ -45,12 +45,14 @@ Semgrep is developed and commercially supported by [r2c, a software security com
...
-| Go         |                            | Kotlin                     |
+| Go         | Kotlin                     | C++                        |
```
Comment (pabloest) [inline docs/index.md:55]:
```suggestion
| Go         | Terraform                  | C++                        |
```

File: docs/status.md [hunk 3]
```diff
@@ -13,11 +13,12 @@ import MoreHelp from "/src/components/MoreHelp"
 
 ## Language maturity
 
+<!--  coupling: if you modify this table, modify also index.md -->
 
 | GA ✅      | Beta 🐛                     | Experimental 🚧            |
 |:---------- |:---------------------------|:---------------------------|
 | C#         | Terraform                  | C                          |
```
Comment (pabloest) [inline docs/status.md:20]:
```suggestion
| C#         | Kotlin                     | C                          |
```

File: docs/status.md [hunk 4]
```diff
@@ -13,11 +13,12 @@ import MoreHelp from "/src/components/MoreHelp"
...
-| Go         |                            | Kotlin                     |
+| Go         | Kotlin                     | C++                        |
```
Comment (pabloest) [inline docs/status.md:21]:
```suggestion
| Go         | Terraform                  | C++                        |
```

**Comment by aryx** [comment]
This will also help https://github.com/returntocorp/semgrep/issues/4300 by adding dots_chaining for Beta

File: docs/index.md [hunk 5]
```diff
@@ -70,4 +73,4 @@ Semgrep is an evolution of [pfff](https://github.com/returntocorp/pfff/), which
 
 ## Development Philosophy
```
Comment (adamkvitek) [inline docs/index.md:74]:
```suggestion
## Understanding Semgrep development philosophy
```

**Comment by aryx** [comment]
Can we merge this @pabloest ?

**Comment by pabloest** [comment]
@aryx Yep! Will do.

**Review by pabloest — APPROVED** [review]

**Pull request merged**